cli/src/main/java/org/apache/any23/cli/Rover.java

```diff
@@ -112,12 +117,16 @@ public String getIdentifier() {
     @Parameter(description = "input IRIs {<url>|<file>}+", converter = ArgumentToIRIConverter.class)
     protected List<String> inputIRIs = new LinkedList<>();
 
-    @Parameter(names = { "-e",
-            "--extractors" }, description = "a comma-separated list of extractors, e.g. rdf-xml,rdf-turtle")
-    private List<String> extractors = new LinkedList<>();
+    @Parameter(names = { "-e", "--extractors" }, description = "a comma-separated list of extractors, "
+            + "e.g. rdf-xml,rdf-turtle, etc. A complete extractor list can be obtained by calling ./any23 extractor --list")
+    private List<String> extractors = new LinkedList<String>() {
+        {
+            addAll(eRegistry.getAllNames());
+        }
+    };
 
     @Parameter(names = { "-f",
-            "--format" }, description = "a comma-separated list of writer factories, e.g. notrivial,nquads")
+            "--format" }, description = "a comma-separated list of writer factories, e.g. json,jsonld,nquads,notrivial,ntriples,trix,turtle,uri")
```

Comment:
@lewismc the --extractors improvement looks good, however the writer example won't work. The only reason `notrivial` can precede `nquads` is that `notrivial` is a delegating writer factory... i.e., it *pre-processes* the triples before passing them to the nquads writer factory.

Reply:
(There is only 1 format, ultimately, that you write the output as, IIRC)